### PR TITLE
Document the `migrations` option

### LINKED ISSUE #19: Migrations
Would be useful to have forward migration support built-in. You define the version when something changed and a function where you can manipulate the store and the correct migrations will be applied according to the installed version on the first run after an update.

Quick braindump API:

```js
const Store = require('electron-store');

const migrations = {
	'1.0.0': store => {
		const old = store.get('old');
		store.set('new', old);
		store.delete('old');
	}
};

const store = new Store({migrations});
```

I'm open to feedback.

### PR DESCRIPTION
**Pull request by rafaelramalho19**
Related with https://github.com/sindresorhus/conf/pull/83

Fixes #19

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -96,6 +96,33 @@ store.set('foo', '1');
...
+       },
+       '1.0.0': store => {
+            store.delete('debug phase');
+            store.set('phase', '1.0');
+       },
+       '1.0.2': store => {
+            store.set('phase', '>1.0');
+       }
+    }
+});
```
Comment (sindresorhus) [inline readme.md]: Use tab-indentation.

File: readme.md [hunk 2]
```diff
@@ -96,6 +96,33 @@ store.set('foo', '1');
...
+Type: `Object`
+
+You can use migrations to perform operations to the store whenever a version is switched.
+
+The `migrations` object should be consisted of a key-value pair of `version`: `handler`.
+
+Example:
+
+```js
+const store = new Conf({
```
Comment (sindresorhus) [inline readme.md]: It's not `Conf`.

File: readme.md [hunk 3]
```diff
@@ -96,6 +96,33 @@ store.set('foo', '1');
 
 **Note:** The `default` value will be overwritten by the `defaults` option if set.
 
+### migrations
+
+Type: `Object`
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
Type: `object`
```

File: readme.md [hunk 4]
```diff
@@ -96,6 +96,33 @@ store.set('foo', '1');
...
+
+You can use migrations to perform operations to the store whenever a version is switched.
```
Comment (sindresorhus) [inline readme.md]: > whenever a version is switched.

I don't think `switched` is the best word to describe this.

Comment (rafaelramalho19) [reply readme.md]: What about upgraded? Makes more sense?

File: readme.md (hunk 3 above)
Comment (rafaelramalho19) [reply readme.md]: The readme on the `conf` project has `Object` in uppercase (and it's the most common convention, although is up to user-preference).

What convention should we use? And If we opt with going with Uppercase Object, should I update this readme type aswell? 😄

File: readme.md (hunk 1 above)
Comment (rafaelramalho19) [reply readme.md]: Alright, will switch. 

But please note that tabs can lead to different indentations depending on the tool being used to preview it.

Please take into consideration: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md010---hard-tabs

File: readme.md (hunk 2 above)
Comment (rafaelramalho19) [reply readme.md]: ooops, nice catch 👍

File: readme.md (hunk 4 above)
Comment (rafaelramalho19) [reply readme.md]: Changed it to "migrated" :)

File: readme.md (hunk 3 above)
Comment (sindresorhus) [reply readme.md]: I used to use `Object`, but I now use exactly the same types as TypeScript, which uses lowercase `object`. So yes, I prefer lowercase everywhere for `object`.

Comment (rafaelramalho19) [reply readme.md]: Alright! I updated every uppercase reference to lowercase :)

**Pull request merged**